internal/nbconn/nbconn_real_non_block_windows.go

```diff
@@ -39,36 +43,48 @@ func setSockMode(fd uintptr, mode sockMode) error {
 	return nil
 }
 
+func (c *NetConn) isDeadlineSet(dl time.Time) bool {
+	return !dl.IsZero() && dl != NonBlockingDeadline && dl != disableSetDeadlineDeadline
```

Comment:
```
return !dl.IsZero() && !dl.Equal(NonBlockingDeadline) && !dl.Equal(disableSetDeadlineDeadline)
```

Reply:
@drakkan is this necessary?  This vars declared as special values here - https://github.com/jackc/pgx/blob/master/internal/nbconn/nbconn.go#L35

And there's also direct comparison with this special values in other parts of code - https://github.com/jackc/pgx/blob/master/internal/nbconn/nbconn.go#L259

Reply:
> @drakkan is this necessary? This vars declared as special values here - https://github.com/jackc/pgx/blob/master/internal/nbconn/nbconn.go#L35
> 
> And there's also direct comparison with this special values in other parts of code - https://github.com/jackc/pgx/blob/master/internal/nbconn/nbconn.go#L259

it does not matter if the location is the same, see [here](https://pkg.go.dev/time#Time.Equal). The best practice is to use Equal

Reply:
@drakkan thanks for the tip, fixed in https://github.com/jackc/pgx/pull/1555/commits/fbe7692f66730900f3a2f9ca589daa66bcd57d6d